test/algorithms/simulation/sidb/critical_temperature.cpp

```diff
@@ -0,0 +1,237 @@
+//
+// Created by Jan Drewniok on 07.02.23.
+//
+
+#include <catch2/catch_template_test_macros.hpp>
+
+#include <fiction/algorithms/simulation/sidb/critical_temperature.hpp>
+#include <fiction/algorithms/simulation/sidb/energy_distribution.hpp>
+#include <fiction/algorithms/simulation/sidb/quicksim.hpp>
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/cell_level_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/hexagonal_layout.hpp>
+#include <fiction/technology/cell_technologies.hpp>
+
+using namespace fiction;
+using namespace detail;
+
+TEMPLATE_TEST_CASE(
+    "Test critical_temperature function", "[critical_temperature]",
+    (cell_level_layout<sidb_technology, clocked_layout<cartesian_layout<siqad::coord_t>>>),
+    (cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<siqad::coord_t, odd_row_hex>>>),
+    (cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<siqad::coord_t, even_row_hex>>>),
+    (cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<siqad::coord_t, odd_column_hex>>>),
+    (cell_level_layout<sidb_technology, clocked_layout<hexagonal_layout<siqad::coord_t, even_column_hex>>>))
+{
+
+    SECTION("empty layout")
+    {
+        TestType lyt{{10, 10}};
+
+        critical_temperature_stats<TestType> criticalstats{};
+        critical_temperature_params const    params{simulation_engine::EXACT,
```

Comment:
For all `const` in this file, please make sure they are `const` West

Reply:
sorry, once again...these things should not happen to me.